src/runner.ts

```diff
@@ -1,17 +1,163 @@
+import * as pt from 'path';
+import * as glob from 'glob';
+import {messages, messagesFormatter} from './messages';
+
+export interface RunOptions {
+    /* Specify a path to folder or a file for the Parasoft coverage report */
+    report: string;
+
+    /* Specify a path to Parasoft tools installation folder or Java installation folder */
+    parasoftToolOrJavaRootPath: string;
+}
 
 export interface RunDetails {
     exitCode: number;
+    convertedCoberturaReportPaths?: string[];
 }
 
 export class CoverageParserRunner {
-    async run() : Promise<RunDetails> {
-        // TODO: Simulate coverageNode input for testing the structure implemented in current task
-        const coberturaCoverage: types.CoberturaCoverage = this.getCoberturaCoverage();
+    async run(runOptions: RunOptions) : Promise<RunDetails> {
+        const parasoftReportPaths = this.findParasoftCoverageReports(runOptions.report);
+        if (!parasoftReportPaths || parasoftReportPaths.length == 0) {
+            return Promise.reject(messagesFormatter.format(messages.coverage_report_not_found, runOptions.report));
+        }
+
+        const javaFilePath = this.getJavaFilePath(runOptions.parasoftToolOrJavaRootPath);
+        if (!javaFilePath) {
+            return { exitCode: -1 }
+        }
+
+        const outcome = await this.convertReportsWithJava(javaFilePath, parasoftReportPaths);
+        if (outcome.exitCode == 0 && outcome.convertedCoberturaReportPaths) {
+            // TODO: Implement Merge multiple cobertura reports
+            // TODO: Implement calculating coverage data from the converted report
+            // TODO: Simulate coverageNode input for testing the structure implemented in current task
+            const coberturaCoverage: types.CoberturaCoverage = this.getCoberturaCoverage();
+
+            await this.generateCoverageSummary(coberturaCoverage);
+        }
+
+        return { exitCode: outcome.exitCode };
+    }
+
+    private findParasoftCoverageReports(reportPath: string) : string[] | undefined {
+        let reportPaths: string[] = [];
+        const workingDir = process.env.GITHUB_WORKSPACE || process.cwd();
+        if (pt.isAbsolute(reportPath)) {
+            core.info(messages.finding_coverage_report);
+            // Handle for path like '/coverage.xml' in Windows
+            reportPath = pt.resolve(reportPath);
+        } else {
+            core.info(messagesFormatter.format(messages.finding_coverage_report_in_working_directory , workingDir));
+            reportPath = pt.join(workingDir, reportPath);
+        }
+
+        // When report path is a directory
+        if (pt.extname(reportPath) == '') {
+            reportPath = pt.join(reportPath, "coverage.xml");
+        }
+
+        reportPath = reportPath.replace(/\\/g, "/");
+
+        try {
+            // Use glob to find the matching report paths
+            reportPaths = glob.sync(reportPath);
+        } catch (error) {
+            return undefined;
```

Comment:
Shouldn't this situation return not just `undefined`, but also include proper error handling? This could be a system error rather than simply returning `undefined` and print that no reports were found.

Reply:
No need to catch the exception, just throw the exception to the top level.